internal/provider/resource_vault_cluster.go

```diff
@@ -0,0 +1,137 @@
+package provider
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/input"
+)
+
+// defaultClusterTimeout is the amount of time that can elapse
+// before a cluster read operation should timeout.
+var defaultVaultClusterTimeout = time.Minute * 5
+
+// createTimeout is the amount of time that can elapse
+// before a cluster create operation should timeout.
+var createVaultClusterTimeout = time.Minute * 35
+
+// deleteTimeout is the amount of time that can elapse
+// before a cluster delete operation should timeout.
+var deleteVaultClusterTimeout = time.Minute * 25
+
+func resourceVaultCluster() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The Vault cluster resource allows you to manage an HCP Vault cluster.",
+		CreateContext: resourceVaultClusterCreate,
+		ReadContext:   resourceVaultClusterRead,
+		DeleteContext: resourceVaultClusterDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create:  &createVaultClusterTimeout,
+			Delete:  &deleteVaultClusterTimeout,
+			Default: &defaultVaultClusterTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"cluster_id": {
+				Description:      "The ID of the HCP Vault cluster.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"hvn_id": {
+				Description:      "The ID of the HVN this HCP Vault cluster is associated to.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// optional fields
+			"public_endpoint": {
+				Description: "Denotes that the cluster has a public endpoint for the Vault UI. Defaults to false.",
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"initial_vault_version": {
+				Description:      "The initial Vault version to use when creating the cluster. Once the cluster is created, this value is no longer used. If not specified, it is defaulted to the version that is currently recommended by HCP.",
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateDiagFunc: validateSemVer,
+				DiffSuppressFunc: func(_, old, new string, _ *schema.ResourceData) bool {
+					return input.NormalizeVersion(old) == input.NormalizeVersion(new)
+				},
+				ForceNew: true,
```

Comment:
I'm probably just missing context, but wanted to ask: do we plan to support version upgrades for Vault through Terraform?

Reply:
Ooh you're right, a customer can't upgrade a cluster yet, so this would always be a ForceNew